app/signer/bls.go

```diff
@@ -140,9 +160,13 @@ func (k *BlsKey) Save(password string) {
 		panic(fmt.Errorf("failed to write BLS key: %w", err))
 	}
 
-	// save used password to file
-	if err := tempfile.WriteFileAtomic(k.passwordPath, []byte(password), 0600); err != nil {
-		panic(fmt.Errorf("failed to write BLS password: %w", err))
+	// Only save the password to a file if it's not provided via environment variable
+	// This maintains backward compatibility while avoiding storing the password in plaintext
+	// when it's provided via environment variable
+	if os.Getenv(BlsPasswordEnvVar) == "" {
```

Comment:
hmm I think this is a bit weird condition tbh 

Imagine I am creating my BLS key 1st time, I will be asked by some sytsem prompt to enter passphrase, I will enter it, my key and passphrase will be created. But at no  point in this flow I have time to set up my env variable, so by default file with it will be created. This is not necessary my intention.

I think when creating key / passphrase it is customary to display prompt for passphrase, which will also state that the passphrase should be remembered by the user (or written somewhere)

We should only write the passphrase to the file if explicity asked by user by, for example, providing optional path to the file.

Wdyt @Vvaradinov @gitferry